### PR TITLE
[Users] Add delete user mutation fixes(#134)

### LINKED ISSUE #134: [Users] Add delete user mutation
- [x] Add a mutation `userDisable` to delete the user.
- [x] The action should only be allowed to admins.
- [x] Update login callback to not create a user if a deleted entity for that user is present

### PR DESCRIPTION
**Pull request by jugshaurya**
## Issue
fixes: #134 

## Solution
 - deletedAt property is used to soft delete the fields and can be used to restore them.
## Checklist
*PR will only be reviewed if the checklist is completed*
- [x] I have done extensive testing for my changes.
- [x] I have self-reviewed my PR and removed all un-necessary changes.
- [x] My changes works well on both desktop and mobile environments.
- [x] I have made sure that my changes solves the actual problem mentioned in the issue.
- [x] The PR contains only 1 commit and have been updated with `master`.

### CODE REVIEW

**Comment by jugshaurya** [comment]
@hereisnaman please review,

File: api/services/User/index.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import Availability from './Loaders/Availability';
 
 export default class User extends BaseModelService {
   static findByUsername(username) {
-    return Models.User.findOne({ where: { username } });
+    return Models.User.findOne({ where: { username }, paranoid: false });
```
Comment (0xnmn) [inline api/services/User/index.js]: expect options as a second arg and spread it. While calling from the login callback pass the paranoid as false from there inside the options object.

**Review by 0xnmn — APPROVED** [review]

**Comment by jugshaurya** [comment]
@hereisnaman, please review. Changes done

File: api/services/User/index.js (hunk 1 above)
Comment (jugshaurya) [reply api/services/User/index.js]: done

**Comment by jugshaurya** [comment]
@hereisnaman  , testing required means, you are going to test it or I am?

**Comment by jugshaurya** [comment]
@hereisnaman, can you help me out how can I test it as no frontend for it is written yet? maybe a new issue should be raised to add this feature with a checklist so that I can add that as well accordingly and test later on! and move on to issue #135 and assign that to me.

**Comment by 0xnmn** [comment]
> @hereisnaman, can you help me out how can I test it as no frontend for it is written yet? maybe a new issue should be raised to add this feature with a checklist so that I can add that as well accordingly and test later on! and move on to issue #135 and assign that to me.

visit localhost:3000/api/graphql in browser and test from there.

**Comment by jugshaurya** [comment]
@hereisnaman Testing please review: 

**Test a:** Add a mutation userDisable to delete the user.
1a
deleted_at got filled
1b

**Test b:**  The action should only be allowed to admins.
*non-admin-viewer*
2a
*unauthorized*
2b

**Test c:** Update login callback to not create a user if a deleted entity for that user is present
How to test this ?
By the way in impersonating form:- once a user is deleted it is not showing there to log in as him/her.

**Comment by 0xnmn** [comment]
@jugshaurya for impersonation and user listing, you'll need to handle few tasks from https://github.com/coding-blocks/gondor/issues/134 

Lets include the last 2 tasks from that issue in this PR.

**Comment by jugshaurya** [comment]
> @jugshaurya for impersonation and user listing, you'll need to handle few tasks from https://github.com/coding-blocks/gondor/issues/134 
> 
> Lets include the last 2 tasks from that issue in this PR.

@hereisnaman, i already included the last 2 tasks from issue https://github.com/coding-blocks/gondor/issues/134 in this PR where if deleted_at property is not null, we redirect to homepage. And for cheking for admins test-b is also passing. You can see that in above screenshots

**Comment by 0xnmn** [comment]
> > @jugshaurya for impersonation and user listing, you'll need to handle few tasks from #134
> > Lets include the last 2 tasks from that issue in this PR.
> 
> @hereisnaman, i already included the last 2 tasks from issue #134 in this PR where if deleted_at property is not null, we redirect to homepage. And for cheking for admins test-b is also passing. You can see that in above screenshots

Then why is " By the way in impersonating form:- once a user is deleted it is not showing there to log in as him/her." this an issue. You can pass the required arguments in the query to display those right.

**Comment by jugshaurya** [comment]
> > @hereisnaman, i already included the last 2 tasks from issue #134 in this PR where if deleted_at property is not null, we redirect to homepage. And for cheking for admins test-b is also passing. You can see that in above screenshots
> 
> Then why is " By the way in impersonating form:- once a user is deleted it is not showing there to log in as him/her." this an issue. You can pass the required arguments in the query to display those right.

@hereisnaman , No that was not an issue, actually, i was saying for the test case c . Once the user is deleted and delete_at is filled, the users that are deleted are not showing in the form , which is what we want. So everything is fine. I was just giving explaination for test case c.
also for the point 3 in the checklist, I just used these two lines 

```js
   await user.loadByUsername(profile.username, { paranoid: false });
    if (user.deleted_at) return App.redirectToHome(res);
```

File: app/mutations/userDisable.graphql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+mutation UserDisable($id: ID!) {
+  disableUser: UserDisable(id: $id) {
```
Comment (0xnmn) [inline app/mutations/userDisable.graphql]: The mutation name is `userDisable`, fix the casing.

**Review by 0xnmn — CHANGES_REQUESTED** [review]

File: app/mutations/userDisable.graphql (hunk 2 above)
Comment (jugshaurya) [reply app/mutations/userDisable.graphql]: oh sorry for that typo

**Comment by jugshaurya** [comment]
@hereisnaman, please review

**Comment by jugshaurya** [comment]
@hereisnaman, require some more changes?

**Review by 0xnmn — APPROVED** [review]

**Pull request merged**